### PR TITLE
Partnerships dashboard + integration of rainbowkit

### PR DESCRIPTION
**Pull request by pernifin**
## PR-Codex overview
This PR adds some new dependencies and introduces several new components and utility functions, as well as updates the global styles and the Tailwind config.

### Detailed summary
- Added `clsx` and `tailwind-merge` dependencies
- Added `ConnectButton`, `Separator`, `Badge`, and `PartnershipCard` components
- Added `cn` utility function
- Updated global styles to include new color variables
- Updated Tailwind config to include new styles and animations

> The following files were skipped due to too many changes: `src/components/ui/card.tsx`, `package-lock.json`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/components/partnershipCard.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  partnership: Partnership;
+}
+
+function TwitterButton({ uri }: { uri: string }) {
+  return (
+    <a href={uri} target="_blank" className="rounded-full bg-gray-100 p-2 text-black">
+      <Twitter size={16} />
+    </a>
+  );
+}
```
Comment (didostap) [inline src/components/partnershipCard.tsx:28]: Let's extract it to a separate file as we gonna reuse it on Proposal details page

File: src/pages/index.tsx [hunk 2]
```diff
@@ -4,22 +4,45 @@ import { serverSideHelpers } from "utils/serverSideHelpers";
...
+
+  const partnerships = Array(10).fill({
+    id: "1",
+    title: "Test Partnership",
+    category: "MARKETING",
+    description: "Lorem ipsum dolor sit amet ".repeat(5),
+    twitterURI: "https://twitter.com/0xWagmi",
+  });
+
+  // const partnerships = data?.pages.flatMap((page) => page.partnerships) ?? [];
```
Comment (didostap) [inline src/pages/index.tsx:24]: rm pls

File: src/components/ui/separator.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+"use client"
+
+import * as React from "react"
+import * as SeparatorPrimitive from "@radix-ui/react-separator"
+
+import { cn } from "~/lib/utils"
+
+const Separator = React.forwardRef<
+  React.ElementRef<typeof SeparatorPrimitive.Root>,
```
Comment (didostap) [inline src/components/ui/separator.tsx:9]: What is the point to pass ref here?

**Review by didostap — COMMENTED** [review]
LGTM!
Just a few improvement proposals

**Review by JonWofr — APPROVED** [review]

**Pull request closed without merging**